plugins/modules/remote_management/redfish/ilo_redfish_info.py

```diff
@@ -0,0 +1,161 @@
+'''
+
+CATEGORY_COMMANDS_ALL = {
+    "Sessions": ["GetiLOSessions"]
+}
+
+CATEGORY_COMMANDS_DEFAULT = {}
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible_collections.community.general.plugins.module_utils.ilo_redfish_utils import iLORedfishUtils
+
+
+def main():
+    result = {}
+    category_list = []
+    module = AnsibleModule(
+        argument_spec=dict(
+            category=dict(required=True, type='list', elements='str'),
+            command=dict(required=True, type='list', elements='str'),
+            baseuri=dict(required=True),
+            username=dict(),
+            password=dict(no_log=True),
+            auth_token=dict(no_log=True),
+            timeout=dict(type='int', default=10)
+        ),
+        required_together=[
+            ('username', 'password'),
+        ],
+        required_one_of=[
+            ('username', 'auth_token'),
+        ],
+        mutually_exclusive=[
+            ('username', 'auth_token'),
+        ],
+        supports_check_mode=True
+    )
+
+    creds = {"user": module.params['username'],
+             "pswd": module.params['password'],
+             "token": module.params['auth_token']}
+
+    timeout = module.params['timeout']
+
+    root_uri = "https://" + module.params['baseuri']
+    rf_utils = iLORedfishUtils(creds, root_uri, timeout, module)
+
+    # Build Category list
+    if "all" in module.params['category']:
+        for entry in CATEGORY_COMMANDS_ALL:
+            category_list.append(entry)
+    else:
+        # one or more categories specified
+        category_list = module.params['category']
+
+    for category in category_list:
+        command_list = []
+        # Build Command list for each Category
+        if category in CATEGORY_COMMANDS_ALL:
+            if not module.params['command']:
+                # True if we don't specify a command --> use default
+                command_list.append(CATEGORY_COMMANDS_DEFAULT[category])
+            elif "all" in module.params['command']:
+                for entry in range(len(CATEGORY_COMMANDS_ALL[category])):
+                    command_list.append(CATEGORY_COMMANDS_ALL[category][entry])
+            # one or more commands
+            else:
+                command_list = module.params['command']
+                # Verify that all commands are valid
+                for cmd in command_list:
+                    # Fail if even one command given is invalid
+                    if cmd not in CATEGORY_COMMANDS_ALL[category]:
+                        module.fail_json(msg="Invalid Command: %s" % cmd)
+        else:
+            # Fail if even one category given is invalid
+            module.fail_json(msg="Invalid Category: %s" % category)
+
+    # Organize by Categories / Commands
+    if category == "Sessions":
+        for command in command_list:
+            if command == "GetiLOSessions":
+                result = rf_utils.get_ilo_sessions()
```

Comment:
You are overwriting the results of all previous commands. That does not really make any sense!